File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [0];
    var leftDays=progresses.map((progress, idx)=> Math.ceil((100-progress)/speeds[idx]));
    var max=leftDays[0];
    var idx=0;

    leftDays.forEach( each => each<=max ? ( answer[idx] += 1 ) : ( max=each, answer[++idx] = 1 ));
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> howlong;
    int day=0;

    for(int i=0;i<progresses.size();i++){
        if((100-progresses[i])%speeds[i]==0)
            day=(100-progresses[i])/speeds[i];
        else
            day= (100-progresses[i])/speeds[i]+1;
        howlong.push_back(day);
    }
    int max=howlong[0];
    int cnt=1;
    for (int i=1;i<howlong.size();i++){
        if(max>=howlong[i]){
            cnt++;
        }else{
            answer.push_back(cnt);
            cnt=1;
            max=howlong[i];
        }
    }
    answer.push_back(cnt);

    return answer;
}